driver/dma_control.c

```diff
@@ -235,11 +258,12 @@ int initialise_dma_control(
     mutex_init(&dma->mutex);
     init_completion(&dma->dma_done);
 
-    reset_dma_controller(dma);
+    TEST_RC(rc=reset_dma_controller(dma), reset_error, "Fail to reset DMA");
```

Comment:
No: please don't use assignment expressions unless forced to!  In this case it's easy to write:
```c
rc = reset_dma_controller(dma);
TEST_RC(rc, reset_error, "Fail to reset DMA");
```

Reply:
Not sure I'm being consistent here, to be honest; the `TEST_...` macros in my other bodies of code (particularly [PandaBlocks-server](https://github.com/PandABlocks/PandABlocks-server) and [epics_device](https://github.com/Araneidae/epics_device) rely on assignment expressions, but we're not using them anywhere else in this driver so I'll stand by this.

Reply:
I already spitted it, though I can roll that back if you prefer so